File: src/main/java/com/temelyan/pomoapp/web/pomo/PomoRestController.java
package com.temelyan.pomoapp.web.pomo;

import com.temelyan.pomoapp.model.Project;
import org.springframework.http.MediaType;
import org.springframework.web.bind.annotation.*;

import java.util.Set;

@RestController
@RequestMapping("/ajax/pomo/")
public class PomoRestController extends AbstractPomoController {
    @PostMapping(path = "add")
    public void addPomo(@RequestParam("length") int length,
                        @RequestParam("taskId") int taskId,
                        @RequestParam("clientTimeZone") int clientTimeZone) {
        super.add(length, taskId, clientTimeZone);
    }

    @GetMapping(path = "get", produces = MediaType.APPLICATION_JSON_VALUE)
    public Set<Project> getUserWithPomosInDateRange(@RequestParam("from") String from, @RequestParam("to") String to) {
        return super.getUserWithPomosInDateRange(from, to);
    }
}

File: src/main/java/com/temelyan/pomoapp/web/pomo/PomoRestController.kt
package com.temelyan.pomoapp.web.pomo

import com.temelyan.pomoapp.model.Project
import com.temelyan.pomoapp.service.AuthorisedUserService
import com.temelyan.pomoapp.service.PomoService
import com.temelyan.pomoapp.service.ProjectService
import org.springframework.http.MediaType
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/pomo/")
class PomoRestController(authorisedUserService: AuthorisedUserService,
                         projectService: ProjectService,
                         pomoService: PomoService)
    : AbstractPomoController(authorisedUserService, projectService, pomoService) {

    @PostMapping(path = ["add"])
    fun addPomo(@RequestParam("length") length: Int,
                @RequestParam("taskId") taskId: Int,
                @RequestParam("clientTimeZone") clientTimeZone: Int) {
        //todo add validation that the task belongs to the authenticated user
        super.add(length, taskId, clientTimeZone)
    }

    @GetMapping(path = ["get"], produces = [MediaType.APPLICATION_JSON_VALUE])
    public override fun getUserWithPomosInDateRange(@RequestParam("from") from: String,
                                                    @RequestParam("to") to: String): Set<Project> {
        return super.getUserWithPomosInDateRange(from, to)
    }
}
